client/asset/eth/nodeclient.go

```diff
@@ -313,38 +332,36 @@ func (n *nodeClient) refund(txOpts *bind.TransactOpts, secretHash [32]byte, serv
 }
 
 func (n *nodeClient) netFeeState(ctx context.Context) (baseFees, tipCap uint64, err error) {
-	bn, err := n.ec.BlockNumber(ctx)
-	if err != nil {
-		return 0, 0, err
-	}
-	blk, err := n.ec.BlockByNumber(ctx, big.NewInt(int64(bn)))
+	hdr, err := n.bestHeader(ctx)
 	if err != nil {
 		return 0, 0, err
 	}
-	base := misc.CalcBaseFee(n.leth.ApiBackend.ChainConfig(), blk.Header())
+
+	base := misc.CalcBaseFee(n.leth.ApiBackend.ChainConfig(), hdr)
 	b := weiToGwei(base)
 	if b == 0 {
 		b = 1
 	}
 
-	tip, err := n.ec.SuggestGasTipCap(ctx)
+	tip, err := n.leth.ApiBackend.SuggestGasTipCap(ctx)
 	if err != nil {
 		return 0, 0, err
 	}
 
-	fmt.Println("--base, tip", base, tip)
-
 	return b, weiToGwei(tip), nil
 }
 
 // getCodeAt retrieves the runtime bytecode at a certain address.
-func (n *nodeClient) getCodeAt(ctx context.Context, contractAddr *common.Address) ([]byte, error) {
-	bn, err := n.ec.BlockNumber(ctx)
+func (n *nodeClient) getCodeAt(ctx context.Context, contractAddr common.Address) ([]byte, error) {
```

Comment:
This one's busted too and I don't know why yet. It's busted if I use the existing method too, but that's not ruling out some other mistake I made.

Reply:
Works fine for me.. maybe try restarting the eth harness?

Reply:
Hmm. I'm still failing. What version of geth are you running?

Reply:
Nevermind. I found the problem.

Reply:
What was wrong?

Reply:
I think I've forgotten now. Sorry I missed this question.